miq_version/__init__.py

```diff
@@ -402,11 +414,17 @@ def parse_template(cls, template_name):
                         template_date = futurecheck(date(year, month, day))
                     except ValueError:
                         continue
-                    dot_version = (version if 'downstream' not in stream_tuple.stream else
-                                   '{}.{}.{}.{}'.format(version[0],
-                                                        version[1],
-                                                        version[2],
-                                                        version[3:]))
+                    if 'downstream' not in stream_tuple.stream:
+                        dot_version = version
+                    elif version.startswith('51'):
+                        dot_version = '{}.{}.{}'.format(version[0],
```

Comment:
I think this should still produce a result with 4 fields, since the regex matches on `(?P<ver>510\d{3})`

Example: 5.10.0.1, 5.10.0.2

Reply:
To be clear, I think the version string formatting should be:

```python
dot_version  = '{}.{}.{}.{}'.format(version[0],
                                                  version[1:3],
                                                  version[3],
                                                  version[4:])

Reply:
agreed, already fixed